### PR TITLE
 Upgrade Prettier to version 1.7.4

### PR DESCRIPTION
**Pull request by jsnajdr**
I rebased the Calypso fork to version 1.7.4 of upstream and reformatted Calypso files with it. Here is th e result.

- the first commit upgrades Prettier in `package.json` and `npm-shrinkwrap.json`.
- the second commit runs `npm run reformat-files`.

There are two major changes since 1.6.1:

**Indenting conditions inside JSX**
Before:
```jsx
<div>
  { something &&
  somethingElse && (
    <ShowSomething />
  ) }
</div>
```
After:
```jsx
<div>
  { something &&
    somethingElse && (
      <ShowSomething />
    ) }
</div>
```
**Fewer parens in ternaries**
Before:
```jsx
something ? (
  translate( 'Something' )
) : (
  translate( 'Other thing' )
)
```
After:
```jsx
something
  ? translate( 'Something' )
  : translate( 'Other thing' )
```

### CODE REVIEW

**Comment by sirreal** [comment]
This should fix an issue we were seeing in #19317

File: client/blocks/comment-button/index.jsx [hunk 1]
```diff
@@ -56,14 +56,14 @@ class CommentButton extends Component {
...
-					<span className="comment-button__label-status">
-						{ translate( 'Comment', 'Comments', {
-							context: 'noun',
-							count: commentCount,
-						} ) }
-					</span>
-				) }
+					commentCount > 0 && (
+						<span className="comment-button__label-status">
+							{ translate( 'Comment', 'Comments', {
```
Comment (a8ci18n) [inline client/blocks/comment-button/index.jsx]: :information_source: This string doesn't exist elsewhere without a **context** (or with a different one). You can probably drop the context. If needed, you can use a translator comment instead to clarify meaning.

**Comment by jsnajdr** [comment]
@sirreal Just to verify my understanding: the version of `calypso-prettier` that we use now references a specific commit in `eslint/typescript-eslint-parser` that doesn't exist any more in the repo?

**Comment by sirreal** [comment]
I've done a visual inspection of a fair amount of the [no-whitespace formatting commit](https://github.com/Automattic/wp-calypso/pull/19519/commits/d8a5cecaf62d4a335eb17e52898a6e114ff780f4?w=1) and it looks good. 👍 

I'll do some smoke testing then accept.

> the version of calypso-prettier that we use now references a specific commit in eslint/typescript-eslint-parser that doesn't exist any more in the repo?

Well, it does exist 😕

https://github.com/eslint/typescript-eslint-parser/commit/801d65585af6995a223136862944095b48f5c567

I'm not really sure what the issue is, I just know that git complains as follows, and that ref appears to be coming from calypso-prettier and is fixed by moving to 1.7.

> reference is not a tree: 801d65585af6995a223136862944095b48f5c567

**Comment by samouri** [comment]
Thanks @jsnajdr for upgrading calypso-prettier!

I actually have one change that [just landed](https://github.com/facebook/jest/pull/4837) in jest that I'd like to update calypso-prettier to take advantage -- would you mind holding off for that?

Note: 1.8 is [soon to be released](https://github.com/prettier/prettier/issues/3117)

**Comment by jsnajdr** [comment]
> would you mind holding off for that?

@samouri I don't mind waiting a few days for the `jest-docblock` changes and for 1.8. The only reason for urgency might be the 'reference is not a tree' issue in #19317.

But maybe a simple update of `package-lock.json` could fix that? There are other [https://github.com/npm/npm/issues/17880](reports) that suggest that it's not limited to Prettier at all. @sirreal? Please don't ask me for any details, I know next to nothing about NPM 5 and its new locking mechanism :)

**Comment by sirreal** [comment]
No rush yet from #19317, there's one other known blocker. I think waiting a few days is 👌

**Comment by samouri** [comment]
I've updated the jest-docblock dep in calypso-prettier, recreated these commits, and force pushed.

**Comment by simison** [comment]
FYI, [1.8 is out](https://github.com/prettier/prettier/releases).

**Comment by sirreal** [comment]
Any blockers for this? @jsnajdr can we move ahead with `1.7` or do you want to go for `prettier@1.8.1`?

**Comment by jsnajdr** [comment]
@sirreal upgrade to 1.8.1 is going to be nontrivial -- I'll need some time to resolve all the conflicts and test everything.

If you want to get new Prettier ASAP, the best thing I can offer right now is Prettier 1.7.4 :) Thoughts?

**Comment by sirreal** [comment]
I'm happy with the `1.7.4` upgrade if we're not waiting on anything else (`jest-docblock`?)

That unblocks 1.5 PRs for me.

> upgrade to 1.8.1 is going to be nontrivial […] Thoughts?

🤔  Just use `prettier` and save us the work :trollface: 😝

**Comment by jsnajdr** [comment]
> Just use prettier and save us the work

I meant the upgrade of the Calypso Prettier fork to 1.8.1 -- I'll need to reapply our Prettier patches to the updated Prettier sources and resolve conflicts.

Reformatting the Calypso sources is a trivial matter, of course 🙂

**Pull request merged**